src/fragments/lib/auth/js/mfa.mdx

```diff
@@ -14,81 +14,184 @@ With TOTP (Time-based One-time Password), your app user is challenged to complet
+  // Then you will have your TOTP account in your TOTP-generating app (like Google Authenticator)
+  // Use the generated one-time password to verify the setup
+  try {
+    const cognitoUserSession = await Auth.verifyTotpToken(user, challengeAnswer);
+    // don't forget to set TOTP as the preferred MFA method
+    await Auth.setPreferredMFA(user, 'TOTP');
+  } catch (error) {
+    // Token is not verified
+  }
 
+  // ...
 
+  // Finally, when sign-in with MFA is enabled, use the confirmSignIn method
+  // to pass the TOTP code and MFA type.
+  await Auth.confirmSignIn(user, code, mfaType); // Optional, MFA Type e.g. SMS_MFA || SOFTWARE_TOKEN_MFA
+}
 ```
+</Block>
+</BlockSwitcher>
 
 ## Setup MFA type
 
 Multiple MFA types supported by Amazon Cognito. You can set the preferred method in your code:
 
-```javascript
+<BlockSwitcher>
+<Block name="TypeScript">
+
+```ts
+import { Auth } from 'aws-amplify';
+
+type SetupMFATypeParameters = {
+  user: string;
+};
+
+export async function setupMFAType({ user }: SetupMFATypeParameters) {
+  // You can select preferred mfa type, for example:
+  // Select TOTP as preferred
+  try {
+    const data = await Auth.setPreferredMFA(user, 'TOTP');
+    console.log(data);
+  } catch (error) {
+    console.log('error setting up MFA type', error);
+  }
+
+  // Select SMS as preferred
+  await Auth.setPreferredMFA(user, 'SMS');
+
+  // Select no-mfa
+  await Auth.setPreferredMFA(user, 'NOMFA');
+}
+```
+</Block>
+<Block name="JavaScript">
+
+```js
 import { Auth } from 'aws-amplify';
 
-// You can select preferred mfa type, for example:
-// Select TOTP as preferred
-Auth.setPreferredMFA(user, 'TOTP')
-  .then((data) => {
+async function setupMFAType(user) {
+  // You can select preferred mfa type, for example:
+  // Select TOTP as preferred
+  try {
+    const data = await Auth.setPreferredMFA(user, 'TOTP');
     console.log(data);
-    // ...
-  })
-  .catch((e) => {});
+  } catch (error) {
+    console.log('error setting up MFA type', error);
+  }
 
-// Select SMS as preferred
-Auth.setPreferredMFA(user, 'SMS');
+  // Select SMS as preferred
+  await Auth.setPreferredMFA(user, 'SMS');
 
-// Select no-mfa
-Auth.setPreferredMFA(user, 'NOMFA');
+  // Select no-mfa
+  await Auth.setPreferredMFA(user, 'NOMFA');
+}
 ```
+</Block>
+</BlockSwitcher>
 
 ## Retrieve current preferred MFA type
 
 You can get current preferred MFA type in your code:
 
-```javascript
+<BlockSwitcher>
+<Block name="TypeScript">
+
+```ts
+import { Auth } from 'aws-amplify';
+
+type GetPreferredMFATypeParameters = {
+  user: string;
+};
+
+export async function getPreferredMFAType({ user }: GetPreferredMFATypeParameters) {
+  // Will retrieve the current mfa type from cache
+  // bypassCache is optional, by default is false.
+  // If set to true, it will get the MFA type from server side instead of from local cache.
+  const data = await Auth.getPreferredMFA(user, { bypassCache: false });
```

Comment:
```suggestion
  // If set to true, it will get the MFA type from server side instead of from local cache.
  const user = Auth.currentAuthenticatedUser();
  const data = await Auth.getPreferredMFA(user, { bypassCache: false });
```